### PR TITLE
Various changes to failure messages

### PR DESCRIPTION
**Pull request by sidkri**

### CODE REVIEW

File: DiagnosticsExtension/Models/ConnectionStringValidator/Constants.cs [hunk 1]
```diff
@@ -14,38 +14,51 @@ namespace DiagnosticsExtension.Models.ConnectionStringValidator
...
         public const string QueueServiceUri = "__queueServiceUri";
         public const string ValidCredentialValue = "managedidentity";
         public const string FullyQualifiedNamespace = "__fullyQualifiedNamespace";
-        public const string ManagedIdentityClientIdNullorEmpty = "The {0}__clientId was not found or is set to a blank value.<a href = 'https://docs.microsoft.com/en-us/azure/azure-functions/functions-reference?tabs=blob#common-properties-for-identity-based-connections' target = '_blank' > Click here to know more</a>";
+        public const string UnknownErrorSummary = "Validation of connection string failed due to an unknown error.";
+        public const string GenericDetailsMessage = "Additional details of the error:";
+        public const string ManagedIdentityTutorial = "Here is a <a href='https://docs.microsoft.com/azure/azure-functions/functions-identity-based-connections-tutorial' target='_blank'>relevant tutorial</a>.";
```
Comment (cachai2) [inline DiagnosticsExtension/Models/ConnectionStringValidator/Constants.cs]:
```suggestion
        public const string ManagedIdentityTutorial = "Refer to the <a href='https://docs.microsoft.com/azure/azure-functions/functions-identity-based-connections-tutorial' target='_blank'>relevant managed identity tutorial</a>.";
```

File: DiagnosticsExtension/Models/ConnectionStringValidator/Constants.cs [hunk 2]
```diff
@@ -14,38 +14,51 @@ namespace DiagnosticsExtension.Models.ConnectionStringValidator
...
+        public const string SystemAssignedAuthFailure = "The system assigned managed identity for this Function App does not have access to the resource configured in '{0}'.  See <a href= 'https://docs.microsoft.com/azure/azure-functions/functions-bindings-storage-blob-trigger#identity-based-connections' target='_blank'>relevant docs</a>. " + ManagedIdentityTutorial;
+        public const string UserAssignedAuthFailure = "The configured user assigned managed identity does not have access to the resource configured in '{0}'.  See <a href= 'https://docs.microsoft.com/azure/azure-functions/functions-bindings-storage-blob-trigger#identity-based-connections' target='_blank'>relevant docs</a>. " + ManagedIdentityTutorial;
+        public const string AuthFailureSummary = "Authentication failure. Credentials in connection string configured in app setting '{0}' are invalid or expired.";
+        public const string AuthFailureBlobStorageDocs = "See See <a href= 'https://docs.microsoft.com/azure/azure-functions/functions-bindings-storage-blob-trigger#connection-string' target='_blank'>relevant docs</a>.";
```
Comment (cachai2) [inline DiagnosticsExtension/Models/ConnectionStringValidator/Constants.cs]:
```suggestion
        public const string AuthFailureBlobStorageDocs = "See <a href= 'https://docs.microsoft.com/azure/azure-functions/functions-bindings-storage-blob-trigger#connection-string' target='_blank'>relevant docs</a>.";
```

File: DiagnosticsExtension/Models/ConnectionStringValidator/Constants.cs [hunk 3]
```diff
@@ -14,38 +14,51 @@ namespace DiagnosticsExtension.Models.ConnectionStringValidator
...
-        public const string ServiceUriEmpty = "ServiceUriEmpty";
-        public const string QueueServiceUriMissing = "The {0}__queueServiceUri was not found or is set to a blank value. <a href= 'https://docs.microsoft.com/en-us/azure/azure-functions/functions-bindings-storage-blob-trigger?tabs=in-process%2Cextensionv5&pivots=programming-language-csharp#identity-based-connections' target='_blank'>Click here to know more</a>";
         public const string QueueServiceUri = "__queueServiceUri";
         public const string ValidCredentialValue = "managedidentity";
         public const string FullyQualifiedNamespace = "__fullyQualifiedNamespace";
-        public const string ManagedIdentityClientIdNullorEmpty = "The {0}__clientId was not found or is set to a blank value.<a href = 'https://docs.microsoft.com/en-us/azure/azure-functions/functions-reference?tabs=blob#common-properties-for-identity-based-connections' target = '_blank' > Click here to know more</a>";
+        public const string UnknownErrorSummary = "Validation of connection string failed due to an unknown error.";
+        public const string GenericDetailsMessage = "Additional details of the error:";
```
Comment (cachai2) [inline DiagnosticsExtension/Models/ConnectionStringValidator/Constants.cs]:
```suggestion
        public const string GenericDetailsMessage = "Additional error details:";
```

File: DiagnosticsExtension/Models/ConnectionStringValidator/Constants.cs [hunk 4]
```diff
@@ -14,38 +14,51 @@ namespace DiagnosticsExtension.Models.ConnectionStringValidator
...
+        public const string ServiceBusFQNSEmptySummary = "The app setting '{0}' has no value. See <a href='https://docs.microsoft.com/azure/azure-functions/functions-bindings-service-bus-trigger#identity-based-connections' target='_blank'>relevant docs</a>. " + ManagedIdentityTutorial;
+        public const string EventHubFQNSEmptySummary = "The app setting '{0}' has no value. See <a href='https://docs.microsoft.com/azure/azure-functions/functions-bindings-event-hubs-trigger#identity-based-connections' target='_blank'>relevant docs</a>. " + ManagedIdentityTutorial;
+        public const string ManagedIdentityClientIdEmptySummary = "The app setting '{0}' has no value.";
+        public const string ManagedIdentityClientIdEmptyDetails = "When the app setting '{0}'" + Credential + " is configured to \"managedidentity\", the clientId of a user assigned managed identity assigned to the Function App is expected in the app setting {0}" + ClientId + ". See <a href='https://docs.microsoft.com/azure/azure-functions/functions-reference#common-properties-for-identity-based-connections' target='_blank'>relevant docs</a>. " + ManagedIdentityTutorial;
```
Comment (cachai2) [inline DiagnosticsExtension/Models/ConnectionStringValidator/Constants.cs:38]: This isn't necessarily true right? Customers should be allowed to use Credential without specifying a ClientID

File: DiagnosticsExtension/Models/ConnectionStringValidator/Constants.cs [hunk 5]
```diff
@@ -14,38 +14,51 @@ namespace DiagnosticsExtension.Models.ConnectionStringValidator
...
+        public const string QueueServiceUriEmptySummary = "The app setting '{0}' has no value. See <a href='https://docs.microsoft.com/azure/azure-functions/functions-bindings-storage-queue-trigger#identity-based-connections' target='_blank'>relevant docs</a>. " + ManagedIdentityTutorial;
+        public const string ServiceBusFQNSEmptySummary = "The app setting '{0}' has no value. See <a href='https://docs.microsoft.com/azure/azure-functions/functions-bindings-service-bus-trigger#identity-based-connections' target='_blank'>relevant docs</a>. " + ManagedIdentityTutorial;
+        public const string EventHubFQNSEmptySummary = "The app setting '{0}' has no value. See <a href='https://docs.microsoft.com/azure/azure-functions/functions-bindings-event-hubs-trigger#identity-based-connections' target='_blank'>relevant docs</a>. " + ManagedIdentityTutorial;
+        public const string ManagedIdentityClientIdEmptySummary = "The app setting '{0}' has no value.";
```
Comment (cachai2) [inline DiagnosticsExtension/Models/ConnectionStringValidator/Constants.cs:37]: Should this have the same doc link as previous lines such as line 36?

File: DiagnosticsExtension/Models/ConnectionStringValidator/Constants.cs [hunk 6]
```diff
@@ -14,38 +14,51 @@ namespace DiagnosticsExtension.Models.ConnectionStringValidator
...
-        public const string StorageAccountResourceNotFound = "The resource specified in the app setting {0} was not found. Please check the value of the setting.";
-        public const string MalformedConnectionStringDetails = "The connection string configured is invalid(e.g.missing some required elements). Please check the value configured in the app setting ";
-        public const string ManagedIdentityCredentialInvalid = "The {0}__credential was not set to 'managedidentity'.<a href = 'https://docs.microsoft.com/en-us/azure/azure-functions/functions-reference?tabs=blob#common-properties-for-identity-based-connections' target = '_blank' > Click here to know more</a>";
-        #endregion
+        public const string ClientIdInvalidTokenGeneratedSummary = "The value of app setting '{0}'" + ClientId + " does not match any user assigned managed identity assigned to this app. See<a href= 'https://docs.microsoft.com/azure/azure-functions/functions-bindings-storage-blob-trigger#identity-based-connections' target='_blank'> relevant docs</a>. " + ManagedIdentityTutorial;
```
Comment (cachai2) [inline DiagnosticsExtension/Models/ConnectionStringValidator/Constants.cs]:
```suggestion
        public const string ClientIdInvalidTokenGeneratedSummary = "The value of the app setting '{0}'" + ClientId + " does not match any user assigned managed identity assigned to this app. See<a href= 'https://docs.microsoft.com/azure/azure-functions/functions-bindings-storage-blob-trigger#identity-based-connections' target='_blank'> relevant docs</a>. " + ManagedIdentityTutorial;
```

File: DiagnosticsExtension/Models/ConnectionStringValidator/Constants.cs (hunk 2 above)
Comment (cachai2) [reply DiagnosticsExtension/Models/ConnectionStringValidator/Constants.cs]: This applies for lines 46-49 as well

File: DiagnosticsExtension/Models/ConnectionStringValidator/Constants.cs [hunk 7]
```diff
@@ -14,38 +14,51 @@ namespace DiagnosticsExtension.Models.ConnectionStringValidator
...
+        public const string AuthFailureFileShareStorageDocs = "See See <a href= 'https://docs.microsoft.com/azure/storage/common/storage-account-keys-manage' target='_blank'>relevant docs</a>.";
+        public const string AuthFailureServiceBusDocs = "See See <a href= 'https://docs.microsoft.com/azure/azure-functions/functions-bindings-service-bus-trigger#connection-string' target='_blank'>relevant docs</a>.";
+        public const string AuthFailureEventHubsDocs = "See See <a href= 'https://docs.microsoft.com/azure/azure-functions/functions-bindings-event-hubs-trigger#connection-string' target='_blank'>relevant docs</a>.";
+        public const string StorageAccessRestrictedDetails = "This can be due to firewall rules on the resource.  Please check if you have configured firewall rules or a private endpoint and that they correctly allow access from the Function App.  See <a href= 'https://docs.microsoft.com/en-us/azure/storage/common/storage-network-security?tabs=azure-portal' target='_blank'>Storage account network security</a> for additional details.";
```
Comment (cachai2) [inline DiagnosticsExtension/Models/ConnectionStringValidator/Constants.cs]:
```suggestion
        public const string StorageAccessRestrictedDetails = "This may be due to firewall rules on the resource.  Please check if you have configured firewall rules or a private endpoint and that they correctly allow access from the Function App.  See <a href= 'https://docs.microsoft.com/en-us/azure/storage/common/storage-network-security?tabs=azure-portal' target='_blank'>Storage account network security</a> for additional details.";
```

Comment (cachai2) [reply DiagnosticsExtension/Models/ConnectionStringValidator/Constants.cs]: Applies to lines 51 & 52 as well. Clarifying, when we say 'This' in the first sentence, we have a previous error message saying connection failed, and this is the more granular error message?

File: DiagnosticsExtension/Models/ConnectionStringValidator/Constants.cs [hunk 8]
```diff
@@ -14,38 +14,51 @@ namespace DiagnosticsExtension.Models.ConnectionStringValidator
...
+        public const string EventHubAccessRestrictedDetails = "This can be due to firewall rules on the resource.  Please check if you have configured firewall rules or a private endpoint and that they correctly allow access from the Function App.  See <a href= 'https://docs.microsoft.com/en-us/azure/event-hubs/network-security' target='_blank'>Event Hubs network security</a> for additional details.";
+        public const string DnsLookupFailed = "The service resource specified in the connection string was not found. Please check the value of the setting.";
+        public const string FQNamespaceResourceNotFound = "The resource specified in the app setting '{0}" + FullyQualifiedNamespace + "' was not found. Please check the value of the setting.";
+        public const string StorageAccountResourceNotFound = "The resource specified in the app setting '{0}' was not found. Please check the value of the setting.";
+        public const string MalformedConnectionStringDetails = "The connection string configured in app setting '{0}' is invalid (e.g.missing some required elements).";
```
Comment (cachai2) [inline DiagnosticsExtension/Models/ConnectionStringValidator/Constants.cs]:
```suggestion
        public const string MalformedConnectionStringDetails = "The connection string configured in app setting '{0}' is invalid. Please double check your connection string";
```

I think it's better to not go into 'missing some required elements' since a customer might not necessarily know what it means to have the correct elements of a connection string

File: DiagnosticsExtension/Models/ConnectionStringValidator/Constants.cs [hunk 9]
```diff
@@ -14,38 +14,51 @@ namespace DiagnosticsExtension.Models.ConnectionStringValidator
...
+        public const string EventHubEntityNotFoundSummary = "The configured Event Hub '{0}' was not found.";
+        public const string EventHubEntityNotFoundDetails = "Refer to <a href='https://docs.microsoft.com/azure/azure-functions/functions-bindings-event-hubs-trigger#attributes' target'_blank'>relevant docs</a> and check the value of the attribute 'EventHubName' in your code.";
+        public const string ServiceBusEntityNotFoundSummary = "The configured Queue or Topic '{0}' was not found.";
+        public const string ServiceBusEntityNotFoundDetails = "Refer to <a href='https://docs.microsoft.com/azure/azure-functions/functions-bindings-service-bus-trigger#attributes' target'_blank'>relevant docs</a> and check the value of the attribute 'QueueName' or 'TopicName' in your code.";
+        public const string ManagedIdentityCredentialInvalidSummary = "The app setting '{0}" + Credential + "' is not valid.  To use identity based connections, set it's value to \"managedidentity\". See <a href='https://docs.microsoft.com/azure/azure-functions/functions-reference#common-properties-for-identity-based-connections' target='_blank'>relevant docs</a>. " + ManagedIdentityTutorial;
```
Comment (cachai2) [inline DiagnosticsExtension/Models/ConnectionStringValidator/Constants.cs]:
```suggestion
        public const string ManagedIdentityCredentialInvalidSummary = "The app setting '{0}" + Credential + "' is not valid.  To use identity-based connections, set its value to \"managedidentity\". See <a href='https://docs.microsoft.com/azure/azure-functions/functions-reference#common-properties-for-identity-based-connections' target='_blank'>relevant docs</a>. " + ManagedIdentityTutorial;
```

File: DiagnosticsExtension/Models/ConnectionStringValidator/Constants.cs [hunk 10]
```diff
@@ -14,38 +14,51 @@ namespace DiagnosticsExtension.Models.ConnectionStringValidator
...
+        public const string KeyVaultReferenceResolutionFailedSummary = "The Azure Key Vault reference configured in app setting '{0}' could not be resolved. See < a href='https://docs.microsoft.com/azure/app-service/app-service-key-vault-references' target='_blank'>relevant docs</a> to correct the issue.";
```
Comment (cachai2) [inline DiagnosticsExtension/Models/ConnectionStringValidator/Constants.cs]:
```suggestion
        public const string KeyVaultReferenceResolutionFailedSummary = "The Azure Key Vault reference configured in app setting '{0}' could not be resolved. See < a href='https://docs.microsoft.com/azure/app-service/app-service-key-vault-references' target='_blank'>relevant docs</a>.";
```

File: DiagnosticsExtension/Models/ConnectionStringValidator/Constants.cs (hunk 1 above)
Comment (sidkri) [reply DiagnosticsExtension/Models/ConnectionStringValidator/Constants.cs]: Trying to keep the message short as it will some times be shown in the summary view instead of the details view.

Changed to "Refer to this relevant tutorial"

File: DiagnosticsExtension/Models/ConnectionStringValidator/Constants.cs (hunk 3 above)
Comment (sidkri) [reply DiagnosticsExtension/Models/ConnectionStringValidator/Constants.cs]: Good catch!  Fixed.

File: DiagnosticsExtension/Models/ConnectionStringValidator/Constants.cs (hunk 2 above)
Comment (sidkri) [reply DiagnosticsExtension/Models/ConnectionStringValidator/Constants.cs]: Fixed.

File: DiagnosticsExtension/Models/ConnectionStringValidator/Constants.cs (hunk 4 above)
Comment (sidkri) [reply DiagnosticsExtension/Models/ConnectionStringValidator/Constants.cs:38]: Unfortunately the underlying SDK or extension currently behaves this way so we are being consistent with that behavior until it is fixed.  The documentation differs from the actual implementation in the SDK based on our testing.

File: DiagnosticsExtension/Models/ConnectionStringValidator/Constants.cs (hunk 5 above)
Comment (sidkri) [reply DiagnosticsExtension/Models/ConnectionStringValidator/Constants.cs:37]: The previous lines don't have a "Details" counterpart which is why I include the docs links in the summary.  In this case, I explain why ClientID is expected which is too wordy to mention in a summary and so I have the details part and this includes the documentation links.  See "ManagedIdentityClientIdEmptyDetails" (line 38)

File: DiagnosticsExtension/Models/ConnectionStringValidator/Constants.cs (hunk 6 above)
Comment (sidkri) [reply DiagnosticsExtension/Models/ConnectionStringValidator/Constants.cs]: Fixed.  Wrote it as I'd say it haha!

File: DiagnosticsExtension/Models/ConnectionStringValidator/Constants.cs (hunk 7 above)
Comment (sidkri) [reply DiagnosticsExtension/Models/ConnectionStringValidator/Constants.cs]: Fixed.  Yes, anytime you see "Details" in the variable name, it always have a corresponding "Summary".  In most cases the prefix is the same but in some cases I have a generic summary like this one.  In this particular case, the summary is inline in the validation logic v/s as a constant which I really should be moved to a constant as well:
ConnectionStringResponseUtility.cs line 139 and 163 and 
response.StatusSummary = "Access to the " + type + " resource is restricted.";

File: DiagnosticsExtension/Models/ConnectionStringValidator/Constants.cs (hunk 9 above)
Comment (sidkri) [reply DiagnosticsExtension/Models/ConnectionStringValidator/Constants.cs]: Fixed

File: DiagnosticsExtension/Models/ConnectionStringValidator/Constants.cs (hunk 8 above)
Comment (sidkri) [reply DiagnosticsExtension/Models/ConnectionStringValidator/Constants.cs]: Changed.

File: DiagnosticsExtension/Models/ConnectionStringValidator/Constants.cs (hunk 10 above)
Comment (sidkri) [reply DiagnosticsExtension/Models/ConnectionStringValidator/Constants.cs]: Changed.  For summary,  shorter is always better :)

**Comment by sidkri** [comment]
The source branch is the latest so closing this PR.  the target branch should be archived.

**Comment by sidkri** [comment]
No further development should be done on this forked repo as the changes are moved to the main repo via PR: https://github.com/Azure/DaaS/pull/65/
